Refactor(eos_cli_config_gen): Deprecation of `type` key from ethernet and port-channel interfaces

Shivani-gslab opened the pull request:
## Change Summary

Deprecating `type` key from ethernet and port-channel interfaces

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

- Deprecating `type` key from ethernet and port-channel interfaces
- Updating the tests
- Fixing the tempate

## How to test
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- N/A

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

Shivani-gslab committed:
Refactor(eos_cli_config_gen): Depreacting type key from ethernet and port-channel interfaces

Shivani-gslab committed:
Updating url in schema

MaheshGSLAB reviewed (commented):
Add the deprecated test in negative_unit_tests of cli gen or eos designs

Shivani-gslab committed:
Adding test for depreacted keys

Shivani-gslab commented:
> Add the deprecated test in negative_unit_tests of cli gen or eos designs

Tests for deprecated keys are already added under `eos_cli_config_gen_deprecated_vars` molecule scenario.
Added some missing tests also.

The model in eos_designs is not updated yet, so no updates on the deprecated key tests there.
Updated the tests under `custom_structured_config..` and `structured_config` keys

ClausHolbechArista commented on ansible_collections/arista/avd/molecule/eos_validate_state/inventory/host_vars/dc2-leaf3b.arista.com.yml:
```diff
@@ -1,28 +1,29 @@
 ---
 # Custom interface structured_config with shutdown key missing to make sure the defaults are handled properly
```
Do we still have some test in validate state using the old structured config format? Since validate state is reading structured config.

Shivani-gslab replied:
Reverted the changes with comment for deprecated data-models

ClausHolbechArista committed:
Merge branch 'devel' into type_deprecation

Shivani-gslab committed:
Reverting changes to eos_validate_state tests with comment

ClausHolbechArista committed:
Merge branch 'devel' into type_deprecation
